File: services/api/app/schemas/station.py
"""Station schemas for serialization."""

from pydantic import BaseModel, ConfigDict


class StationResponse(BaseModel):
    """Schema representing a railway station."""

    station_code: str
    station_name: str
    station_type: str | None = None
    block_station: bool | None = None
    ibp: bool | None = None
    flag_station: bool | None = None
    halt: bool | None = None
    platform_available: bool | None = None
    latitude: float | None = None
    longitude: float | None = None
    division: str | None = None
    zone: str | None = None

    model_config = ConfigDict(from_attributes=True)


class StationDetailResponse(StationResponse):
    """Detailed schema representing a station with administrative metadata."""

    out_of_division_station: bool | None = None
    administrative_division: str | None = None
    scope_note: str | None = None
    source_type: str | None = None

File: services/web/src/lib/types/station.ts
export interface Station {
  station_code: string;
  station_name: string;
  station_type: string | null;
  block_station: boolean | null;
  ibp: boolean | null;
  flag_station: boolean | null;
  halt: boolean | null;
  platform_available: boolean | null;
  latitude: number | null;
  longitude: number | null;
  division: string | null;
  zone: string | null;
  out_of_division_station: boolean | null;
  administrative_division: string | null;
  scope_note: string | null;
  source_type: string | null;
}
